modules/addons/src/gltf/gltf-animator.js

```diff
@@ -64,6 +64,38 @@ function applyTranslationRotationScale(gltfNode, node) {
   }
 }
 
+function linearInterpolate(path, t, a, b) {
+  // TODO: implement SLERP when path is rotation
+  return a.map((v, index) => v * (1 - t) + b[index] * t);
+}
+
+function interpolate(time, {input, interpolation, output}, path) {
+  const maxTime = input[input.length - 1];
+  const animationTime = time % maxTime;
+
+  const nextIndex = input.findIndex(t => t >= animationTime);
+  const previousIndex = Math.max(0, nextIndex - 1);
+
+  switch (interpolation) {
+    case 'STEP':
+      return output[previousIndex];
+
+    case 'LINEAR':
+      const previousTime = input[previousIndex];
+      const nextTime = input[nextIndex];
+      const t = (animationTime - previousTime) / (nextTime - previousTime);
```

Comment:
I'd rather see this as:
```
if (nextTime !== previousTime) {
     const t = (animationTime - previousTime) / (nextTime - previousTime);
     return linearInterpolate(path, t, output[previousIndex], output[nextIndex]);
}
```
It's clearer about what the condition is and doesn't perform the calculations when they won't be used.

Reply:
`if (nextTime > previousTime)`